server/budgettracker.web/Controllers/BudgetApiController.cs

```diff
@@ -21,9 +24,26 @@ public BudgetApiController(IBudgetApi budgetApi)
         }
 
         [HttpPost("create")]
-        public async Task<ApiResponse> CreateBudget(ApiRequest request)
+        public async Task<IActionResult> CreateBudget(ApiRequest request)
         {
-            return await _budgetApi.CreateBudget(request);
+            if(!request.Arguments<CreateBudgetRequestContract>().IsValid())
+            {
+                return BadRequest();
+            }
+
+            try
+            {
+                return Ok(await _budgetApi.CreateBudget(request));    
+            }
+            catch (Exception ex) when (ex.InnerException is AuthenticationException)
+            {
+                if (ex is AuthenticationException) 
```

Comment:
Doesn't the `when (ex.InnerException is AuthenticationException)` do this already?